src/components/Login/Login.js

```diff
@@ -25,29 +25,47 @@ class Login extends Component {
       tokenValid: false,
     };
 
+    this.hasToken();
+  }
+
+  hasToken() {
     AsyncStorage.getItem('token', (err, result) => {
-      if (result)
+      if (result) {
         UserUtil.getMyProfile(this.onTokenValidCheck.bind(this));
-      else
+      } else {
         this.setState({ loaded: true });
+      }
```

Comment:
What about
if (err) {
  this.setState({ loaded: true });
} else if (result) {
  UserUtil.getMyProfile(this.onTokenValidCheck.bind(this));
}